### DISCUSSION TITLE
Using an existing <Redis Connection> for FastAPI plug-in, RedisRouter()

### DISCUSSION THREAD

**Discussion by farzbood** [Lancetnik/Propan#129] [Questions]
Hi everyone,
How can we pass an existing **Redis Connection** to the **RedisRouter()** when using **FastAPI plug-in** ?
Currently it just gets a fresh **url** and/or host, port, db, ... to establish it's own connection.
This is needed specifically when we want to use a **mock redis-server** for **development stage, test purposes** rather than an actual instance.
In our case, such a fake-server provided by https://github.com/cunla/fakeredis-py

Keep up with outstanding work!
PEACE V

**Comment by Lancetnik** [comment]
@farzbood thanks for you feedback. Unfortunately, there is interface to make it yet.
But it is a good idea and I will add it in the nearest future.
For now you can use this dirty trick to set up your connection

```python
from fastapi import FastAPI
from propan.fastapi import RedisRouter
from redis.asyncio import from_url

router = RedisRouter()
router.broker._connection = from_url("redis://localhost:6379")

@router.event("test")
async def handle(msg):
    print(msg)

@router.after_startup
async def test(app):
    await router.broker.publish("test", "test")

app = FastAPI(lifespan=router.lifespan_context)
app.include_router(router)
```

**Reply by farzbood** [reply]
The "dirty trick" covers the case ;)
Thanks

**Comment by Lancetnik** [comment]
By the way, using RedisRouter, you can patch inner broker by TestRedisBroker to test your application locally

```python
router = RedisRouter()
router.broker = TestRedisBroker(router.broker)
```

**Reply by farzbood** [reply]
Well performed.
All the best V
